app/controllers/support_interface/provider_users_controller.rb

```diff
@@ -26,10 +34,10 @@ def edit
 
     def update
       provider_user = ProviderUser.find(params[:id])
+      @form = ProviderUserForm.from_provider_user(provider_user)
+      @form.assign_attributes provider_user_params
 
-      @form = ProviderUserForm.new(provider_user_params.merge(provider_user: provider_user))
-
-      if @form.save
+      if @form.save!
```

Comment:
`save!` will raise `RecordInvalid` rather than return false — I don't think we handle that?

Reply:
`save!` is misleading, because I added a guard checking the model is valid before attemptint to save. Renaming to just `save`.